kernel/aster-nix/src/net/socket/ip/options.rs

```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::impl_socket_options;
+
+impl_socket_options!(
```

Comment:
`IpOptions` seems to be defined but never used, right? Would you like to add these options to both TCP socket and UDP socket in this PR?

Reply:
Are you saying that when the user `setsockopt`, we create a tcp/stream or udp/dgram option instead?
The `ping` from busybox is actually using RAW sockets
```
domain = AF_INET, sock_type = SOCK_RAW, sock_flags = (empty), protocol = IPPROTO_ICMP
```
So as the raw socket is a necessity, we should also have a option defined for the raw socket?

Reply:
I just mean the stream socket should also have IP options. Since stream socket is also based on IP protocol.

https://github.com/asterinas/asterinas/blob/c04fd0c9a0dffc54d2cfc985583569bdc7b7f629/kernel/aster-nix/src/net/socket/ip/stream/mod.rs#L67-L70

It should be refactored as
```rust
struct OptionSet { 
     socket: SocketOptionSet, 
     ip: IpOptionSet,
     tcp: TcpOptionSet, 
 } 
```

Reply:
Oh I got it. Actually the options stuff is really chaotic in a way. Like dgram socket don't even have the sock option setting.

Reply:
Yes, we only cover options in stream socket now. This problem needs to be fixed in the future.

Reply:
Several gvisor syscall test cases involve getting and setting these option values. Therefore, adding these options to `StreamSocket` can be beneficial for passing these test cases(maybe in the future) and ensuring that the default values of these options align with Linux.

On the other hand, if these newly-added `IpOptions` are not utilized elsewhere, it is generally considered poor practice to include them in a PR.

Reply:
With the standard developing process, I guess I should first get the raw socket working and handle corresponding the socket options. Bring in the ipoptions for other socket is great and definitely necessary. But it also takes time do figure out what is missing and patch things up.
However, for now I just want to get the `ping` working, in this case it is more likely a top-down approach to see what is still missing for the `ping`. Making it available on other types of socket is not my top most focus.

Reply:
If you want enable `ping` from busybox, the options introduced by this PR are never used. So they will not be added to raw socket yet.

I have checked the syscalls of `ping`. Here are the results of `strace busybox ping 127.0.0.1`.

```plain
socket(AF_INET, SOCK_RAW, IPPROTO_ICMP) = 3
dup2(3, 0)                              = 0
close(3)                                = 0
setsockopt(0, SOL_SOCKET, SO_BROADCAST, [1], 4) = 0
setsockopt(0, SOL_SOCKET, SO_RCVBUF, [7280], 4) = 0
rt_sigaction(SIGINT, {sa_handler=0x55f602, sa_mask=[INT], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0x41e540}, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=0}, 8) = 0
sendto(0, "\10\0\343C\361(\0\0\356g5+\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 64, 0, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("127.0.0.1")}, 28) = 64
rt_sigaction(SIGALRM, {sa_handler=0x55f889, sa_mask=[ALRM], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0x41e540}, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=0}, 8) = 0
setitimer(ITIMER_REAL, {it_interval={tv_sec=0, tv_usec=0}, it_value={tv_sec=1, tv_usec=0}}, NULL) = 0
recvfrom(0, "E\0\0TB\243@\0@\1\372\3\177\0\0\1\177\0\0\1\10\0\343C\361(\0\0\356g5+"..., 192, 0, {sa_family=AF_INET, sin_port=htons(0), sin_addr=inet_addr("127.0.0.1")}, [16]) = 84
```